solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBHttp2SolrClient.java

```diff
@@ -325,9 +338,24 @@ public LBHttp2SolrClient.Builder setAliveCheckInterval(int aliveCheckInterval) {
       return this;
     }
 
+    /**
+     * Expert Method
+     *
+     * @param queryParams set of param keys that are only sent via the query string. Note that the
+     *     param will be sent as a query string if the key is part of this Set or the SolrRequest's
+     *     query params.
+     * @see org.apache.solr.client.solrj.SolrRequest#getQueryParams
+     */
+    public LBHttp2SolrClient.Builder withQueryParams(Set<String> queryParams) {
```

Comment:
I didn't know about this thing before.  The javadocs are nice but I nonetheless find this confusing; I previously might have thought this was something like default SolrParams, but no.  The odd thing is that this doesn't hold the actual parameters, instead it's _names_ of parameters that only result in some change if the name-value is specified on the SolrRequest.    If this were `withUrlQueryParams` (my preference) or `withQueryStringParams`, it'd be clearer.   WDYT?

Reply:
Yeah, I don't like the name either, and the confusion is real, witness the test that had `key=value` being used as query param!  instead of just `key`.    Any thoguhts on a name that really focuses on the fact that it's the key?   `withUrlQueryKeys` ???    or `withQueryKeys` ?

Reply:
in the spirit of the Builder reading like English more naturally, how about `withTheseParamNamesInTheUrl`.  Yes it's long but who cares -- this is some expert method.

Reply:
ROTFL...   But I can totally get behind that!

Reply:
as you are showing willingness to change the name of fields in this PR (i.e. http2SolrClient), I would also really appreciate any field/parameter named "queryParams" to be something like "urlParamNames" so we are less likely to be confused at a glance with SolrParams.